site/app/models/Email.php

```diff
@@ -14,34 +14,52 @@
  * @method string   getSubject()
  * @method string   getBody()
  * @method string   getUserId()
+ * @method string   getEmail()
  */
 class Email extends AbstractModel {
     /** @prop
-     * @var string Subject line of email */
+     * @var string Subject line of email
+     */
     protected $subject;
     /** @prop
-     * @var string Body of email */
+     * @var string Body of email
+     */
     protected $body;
     /** @prop
-     * @var string user name */
+     * @var string username of student.
+     * Alternative option to providing an email address and to_name.
+     */
     protected $user_id;
+    /** @prop
+     * @var string Email address.
+     * Alternative option to providing a user_id. Should use to_name as well.
+     */
+    protected $email_address;
+    /** @prop
+     * @var string Name of who we're sending to.
+     * Alternative option to providing a user_id.
+     */
+    protected $to_name;
 
-
-  /**
-   * Email constructor.
-   *
-   * @param Core  $core
-   * @param array $details
-   */
-
+    /**
+     * Email constructor.
+     * details must contain a subject, a body, and a user id or email address to send to.
+     *
+     * @param Core  $core
+     * @param array $details
+     */
     public function __construct(Core $core, array $details = []) {
         parent::__construct($core);
         if (count($details) == 0) {
             return;
         }
-        $this->setUserId($details["to_user_id"]);
-        $this->setSubject($this->formatSubject($details["subject"]));
-
+        if (array_key_exists("to_user_id", $details)) {
```

Comment:
You could alternatively use `isset()` here.  This isn't a requested change, just a suggestion.

```suggestion
        if (isset($details["to_user_id"])) {
```